src/QirRuntime/samples/StandaloneInputReference/qir-driver.cpp

```diff
@@ -0,0 +1,126 @@
+            simulatorOutputFile,
+            "File to write the output of the simulator to");
+
+        std::string operationOutputFile;
+        CLI::Option *operationOutputFileOpt = app.add_option(
+            "-o,--operation-output",
+            operationOutputFile,
+            "File to write the output of the Q# operation to");
+
+        // Add the options that correspond to the parameters that the QIR entry-point needs.
+        // Option for a Q# Int type.
+        int64_t anInt = 0;
+        CLI::Option *anIntOpt = app.add_option(
+            "--anint",
+            anInt,
+            "An integer");
+
+        anIntOpt->required();
+
+        // Option for a Q# Double type.
+        double_t aDouble = 0.0;
+        CLI::Option *aDoubleOpt = app.add_option(
+            "--adouble",
+            aDouble,
+            "A double");
+
+        aDoubleOpt->required();
+
+        // Option for a Q# Bool type.
+        bool aBool = false;
+        CLI::Option *aBoolOpt = app.add_option(
+            "--abool",
+            aBool,
+            "A bool");
+
+        aBoolOpt->required();
+
+        // TODO: Add option for Q# Pauli type.
+        // TODO: Add option for Q# Result type.
+        // TODO: Add option for Q# String type.
+
+        // Option for a Q# Array<Int> type.
+        std::vector<int64_t> anIntegerArray;
+        CLI::Option *anIntegerArrayOpt = app.add_option(
+            "--anintegerarray",
+            anIntegerArray,
+            "An integer array");
+
+        anIntegerArrayOpt->required();
+
+        // TODO: Add option for Q# Tuple<T> type.
+
+        // With all the options added, parse arguments from the command line.
+        CLI11_PARSE(app, argc, argv);
+
+        // Redirect the simulator output from std::cout if the --simulator-output option is present.
+        std::ostream* simulatorOutputStream = &std::cout;
+        std::filebuf simulatorOutputFileBuffer;
+        if (!simulatorOutputFileOpt->empty()){
+            simulatorOutputFileBuffer.open(simulatorOutputFile, std::ios::out);
+            std::ostream simulatorOutputFileStream(&simulatorOutputFileBuffer);
+            // TODO: Call into the QIR runtime API to redirect the output of the simulator.
+            simulatorOutputStream = &simulatorOutputFileStream;
+        }
+
+        // Redirect the Q# opertion output from std::cout if the --operation-output option is present.
+        std::ostream* operationOutputStream = &std::cout;
+        std::filebuf operationOutputFileBuffer;
+        if (!operationOutputFileOpt->empty()){
+            operationOutputFileBuffer.open(operationOutputFile, std::ios::out);
+            std::ostream operationOutputFileStream(&operationOutputFileBuffer);
+            operationOutputStream = &operationOutputFileStream;
+        }
+
+        // TODO: Remove this after the Message Q# function is integrated into the QIR runtime.
+        (*operationOutputStream) << "SIMULATOR OUTPUT\n----------------" << std::endl;
```

Comment:
It's been integrated now, right?

Reply:
Yes, this code has been updated to call into the `Quantum::Qis::Internal::SetOutputStream` API.